app/main/views/frameworks.py

```diff
@@ -136,7 +136,23 @@ def update_framework(framework_slug):
             enforce_required=True
         )
 
+    # We can ensure that some of the timestamps we have for a framework are accurate, as they are tied to state changes.
+    status_timestamp_audit_data = {}
+    if json_framework.get('clarificationQuestionsOpen') is False and framework.clarification_questions_open is True:
+        framework.clarifications_close_at_utc = datetime.datetime.utcnow()
+        status_timestamp_audit_data = {"clarifications_close_at_utc set": "clarification questions closed"}
+
+    if json_framework.get('status') == 'live' and framework.status != 'live':
+        framework.framework_live_at_utc = datetime.datetime.utcnow()
+        status_timestamp_audit_data = {"framework_live_at_utc set": "framework status set to 'live'"}
+
+    if json_framework.get('status') == 'expired' and framework.status != 'expired':
+        framework.framework_expires_at_utc = datetime.datetime.utcnow()
+        status_timestamp_audit_data = {"framework_expires_at_utc set": "framework status set to 'expired'"}
+
     if json_framework.get('status', None) == 'pending' and framework.status != 'pending':
```

Comment:
These could probably be an `if/elif/elif` change rather than distinct `if`s

Reply:
Also one of these sets a distinct `None` default value while the rest don't

Reply:
Yeah, that was already there for the supplier details injection stuff. But I don't think it's necessary. `None` is the default return value for `.get()`.

Reply:
And yep, the statuses can if `if/elif`, although as the status can only ever be one thing it seems unnecessary?

Reply:
That's kinda why it's more appropriate to do `if/elif` - there's no sense having them as distinct statements because it lightly implies that they could all match in some circumstance.